core/store/src/trie/update.rs

```diff
@@ -147,6 +150,7 @@ impl TrieUpdate {
 
     pub fn rollback(&mut self) {
         self.prospective.clear();
+        self.contract_storage.rollback();
```

Comment:
`rollback` reverts single receipt/transaction, but contract storage `rollback` clears all accesses/deploys accumulated through chunk execution, so that looks incorrect

Reply:
ok, I missed that. I reorganized `contract.rs` to set up a similar structure to `TrieUpdate` to track uncommitted and committed deploys/calls.